src/app/external-cluster-wizard/steps/external-cluster/provider/eks/component.ts

```diff
@@ -175,16 +210,19 @@ export class EKSClusterSettingsComponent
         .valueChanges.pipe(debounceTime(this._debounceTime))
         .pipe(takeUntil(this._unsubscribe))
         .subscribe(_ => {
-          const vpc = this.controlValue(Controls.Vpc)?.main;
+          this._clearSubnet();
+          this._clearSecurityGroups();
+
+          const vpc = this.controlValue(Controls.Vpc);
           if (vpc) {
             this._onVPCSelectionChange(vpc);
-            this.control(Controls.SubnetIds).enable();
-            this.control(Controls.SecurityGroupsIds).enable();
+            this._enable(true, Controls.SubnetIds);
+            this._enable(true, Controls.SecurityGroupsIds);
           } else {
             this.control(Controls.SubnetIds).disable();
-            this.control(Controls.SubnetIds).setValue([]);
             this.control(Controls.SecurityGroupsIds).disable();
-            this.control(Controls.SecurityGroupsIds).setValue([]);
+            this._enable(false, Controls.SubnetIds);
+            this._enable(false, Controls.SecurityGroupsIds);
```

Comment:
maybe instead of the if else statement we can use this 
```suggestion
          if (vpc) {
            this._onVPCSelectionChange(vpc);
            } 
            this._enable(vpc, Controls.SubnetIds);
            this._enable(vpc, Controls.SecurityGroupsIds);```